torchseg/encoders/timm.py

```diff
@@ -0,0 +1,144 @@
+import timm
+import torch.nn as nn
+from einops import rearrange
+
+from .supported import TIMM_ENCODERS
+
+
+class TimmEncoder(nn.Module):
+    def __init__(
+        self,
+        name,
+        pretrained=True,
+        in_channels=3,
+        depth=None,
+        indices=None,
+        output_stride=32,
+        **kwargs,
+    ):
+        super().__init__()
+
+        assert (
+            depth is not None or indices is not None
+        ), "Either `depth` or `indices` should be specified"
+
+        if indices is not None:
+            depth = len(indices)
+        else:
+            indices = tuple(range(depth))
+
+        total_depth = TIMM_ENCODERS[name.split(".")[0]]["indices"]
+        if len(total_depth) < depth:
+            err = f"""
+            The specified depth={depth}or indices={indices} is greater than
+            the maximum available depth={total_depth} for the {name} encoder.
+            """
+            raise ValueError(err)
+
+        params = dict(
+            in_chans=in_channels,
+            features_only=True,
+            pretrained=pretrained,
+            out_indices=indices,
+            output_stride=output_stride,
+        )
```

Comment:
Maybe we can add the custom `norm_layer` and `act_layer` directly here in `params` (and in `__init__`) if specified instead of leaving them in kwargs, just to make it clearer, what do you think? I just don't know how pretrained weights are handled in case of different norm and activation layers from the usual `BatchNorm2d` and `ReLU`, maybe we could force `pretrained` to False for now

Reply:
Agreed I think this is a good addition. I think the only thing I need to check is just to make sure that act_layer and norm_layer are valid for most timm models. Can also pass these to the decoder as well.

Reply:
Yes, forgot about the decoder.

I think most of timm models accept those parameters (except for old hard-coded models, like xception for instance, and specific transformer architectures which support only norm_layer), it could be a good idea to create a `supported_list` for models accepting both `act_layer` and `norm_layer` to avoid errors, or maybe directly add those to [torchseg/encoders/supported.py](https://github.com/isaaccorley/torchseg/pull/3/files#diff-63d2605567afb123ce55163d6d8f3e7b952d0fbcd9a1cda6b4e94131594c8c01) or to the json file created in [scripts/list_compatible_encoders.py](https://github.com/isaaccorley/torchseg/pull/3/files#diff-a6274f2a450efd1913eaf5f50a6bca64f6ac99dd30cb06351b458d3c9150cd7d)

Reply:
I think I'm going to punt on this for a future PR. Don't want this one to add too many new features than it already has.